test/test_utils.h

```diff
@@ -427,6 +427,17 @@ namespace test_utils {
 		}
 	}
 
+	struct set_test_env {
+		set_test_env(const std::string& evn, const std::string val) {
+			setenv(evn.c_str(), val.c_str(), 1);
```

Comment:
It looks like `setenv` does not exist on Windows, so this is not portable unfortunately. Also if you want this to be a generic utility (which I assume as it is in `test_utils.h`) I think a more robust solution would be to restore the previous value upon destruction.